File: frontend/src/labels.ts
/** Spanish UI labels for the language-neutral keys the API stores. */

import type { ExpiryStatus, Location } from '@/services/types'

export const LOCATION_LABELS: Record<Location, string> = {
  fridge: 'Refrigerador',
  freezer: 'Congelador',
  pantry: 'Alacena',
}

export const STATUS_LABELS: Record<ExpiryStatus, string> = {
  fresh: 'En buen estado',
  expiring_soon: 'Por caducar',
  expired: 'Caducado',
}

/**
 * Phrase a day count the way a person would say it.
 *
 * "Caduca en 0 días" is how a machine talks; the whole point of the list is
 * being scannable at a glance.
 */
export function expiryPhrase(daysUntilExpiry: number): string {
  if (daysUntilExpiry < 0) {
    const days = Math.abs(daysUntilExpiry)
    return days === 1 ? 'Caducó ayer' : `Caducó hace ${days} días`
  }
  if (daysUntilExpiry === 0) return 'Caduca hoy'
  if (daysUntilExpiry === 1) return 'Caduca mañana'
  return `Caduca en ${daysUntilExpiry} días`
}

/** Quantity and unit as one string, skipping the unit when there is none. */
export function quantityLabel(quantity: string, unit: string | null): string {
  // The API sends NUMERIC(10,2) as a string, so whole amounts arrive as "2.00".
  // Trim trailing zeros without touching meaningful decimals: 2.00 -> 2,
  // 1.50 -> 1.5, 1.25 -> 1.25.
  const amount = quantity.includes('.') ? quantity.replace(/0+$/, '').replace(/\.$/, '') : quantity
  return unit ? `${amount} ${unit}` : amount
}

File: backend/app/labels.py
"""Spanish labels for the language-neutral keys the database stores.

Mirrors frontend/src/labels.ts. Duplicated rather than shared because the two
runtimes cannot import each other, and the alert has to speak the same Spanish
the UI does — seeing "fridge" in a notification from an app that says
"Refrigerador" everywhere else would read as a bug.
"""

LOCATION_LABELS: dict[str, str] = {
    "fridge": "Refrigerador",
    "freezer": "Congelador",
    "pantry": "Alacena",
}


def expiry_phrase(days_until_expiry: int) -> str:
    """Phrase a day count the way a person would say it."""
    if days_until_expiry < 0:
        days = abs(days_until_expiry)
        return "caducó ayer" if days == 1 else f"caducó hace {days} días"
    if days_until_expiry == 0:
        return "caduca hoy"
    if days_until_expiry == 1:
        return "caduca mañana"
    return f"caduca en {days_until_expiry} días"
